odoo/addons/test_read_group/tests/test_groupby_week.py

```diff
@@ -62,73 +64,124 @@ def setUpClass(cls):
             {'date': '2022-06-20'},       # W25, W25, W26
         ])
 
+    def add_datetime_values(self):
+        tzinfo = timezone(self.Model._context['tz'])
+        for record in self.records:
+            local_dt = tzinfo.localize(datetime.combine(record.date, time.min))
+            record.datetime = local_dt.astimezone(UTC).replace(tzinfo=None)
+
     def test_belgium(self):
         """ fr_BE - first day of the week = Monday """
+        self.Model = self.Model.with_context(lang='fr_BE', tz='Europe/Brussels')
```

Comment:
Okay, this basically sets the context for the test.  And you also have to call `add_datetime_values()` to set up the data properly.  Why don't you make a method for both?
```suggestion
        self.set_context('fr_BE', 'Europe/Brussels')
```
where `set_context()` is defined as
```py
def set_context(self, lang, tz):
    self.Model = self.Model.with_context(lang=lang, tz=tz)
    # set the datetime values according to date and the given context
    tzinfo = timezone(self.Model._context['tz'])
    for record in self.records:
        local_dt = tzinfo.localize(datetime.combine(record.date, time.min))
        record.datetime = local_dt.astimezone(UTC).replace(tzinfo=None)
```

Reply:
Good point. Initially had a `setUp` doing those based on test name, which felt a bit hacky, but this is a way better approach.